src/Core/Form/IdentifiableObject/CommandBuilder/Accessor/CommandAccessor.php

```diff
@@ -0,0 +1,165 @@
+            try {
+                $value = $this->propertyAccessor->getValue($data, $commandField->getDataPath());
+                $castedValue = $this->castValue($value, $commandField->getType());
+
+                $command = $this->getAppropriateCommand($data, $commandField, $singleStoreCommand, $allStoresCommand);
+                $this->propertyAccessor->setValue($command, $commandField->getCommandSetter(), $castedValue);
+                if (!in_array($command, $modifiedCommands)) {
+                    $modifiedCommands[] = $command;
+                }
+            } catch (NoSuchIndexException $e) {
+                // Data has no value for this field
+            }
+        }
+
+        // Make sure the order of returned commands is always consistent (single store comes first)
+        $commands = [];
+        if (in_array($singleStoreCommand, $modifiedCommands)) {
+            $commands[] = $singleStoreCommand;
+        }
+        if (in_array($allStoresCommand, $modifiedCommands)) {
+            $commands[] = $allStoresCommand;
+        }
+
+        return $commands;
+    }
+
+    /**
+     * Check if the data has a mapping checkbox to modify all stores for the tested field, if so use the allStoresCommand
+     *
+     * @param array $data
+     * @param CommandField $commandField
+     * @param mixed $singleStoreCommand
+     * @param mixed|null $allStoresCommand
+     *
+     * @return mixed
+     */
+    private function getAppropriateCommand(
+        array $data,
+        CommandField $commandField,
+        $singleStoreCommand,
+        $allStoresCommand
+    ) {
+        if (null === $allStoresCommand || !$commandField->isMultistoreField()) {
+            return $singleStoreCommand;
+        }
+
+        $dataPath = $commandField->getDataPath();
+        $lastElement = $dataPath->getElement($dataPath->getLength() - 1);
+        $modifyAllNamePrefix = $this->config->getModifyAllNamePrefix() . $lastElement;
+
+        // Replace last element
+        $stringPath = (string) $dataPath;
+        if (($pos = strrpos($stringPath, $lastElement)) !== false) {
+            $stringPath = substr_replace($stringPath, $modifyAllNamePrefix, $pos, strlen($lastElement));
+        }
+
+        try {
+            $modifyAll = $this->propertyAccessor->getValue($data, $stringPath);
+
+            return $modifyAll ? $allStoresCommand : $singleStoreCommand;
+        } catch (NoSuchIndexException | NoSuchPropertyException $e) {
+            return $singleStoreCommand;
+        }
+    }
+
+    /**
+     * @param mixed $value
+     * @param string $type
+     *
+     * @return bool|int|mixed|string|array
+     */
+    private function castValue($value, string $type)
+    {
+        switch ($type) {
+            case CommandField::TYPE_STRING:
+                return (string) $value;
+            case CommandField::TYPE_BOOL:
+                return (bool) $value;
+            case CommandField::TYPE_INT:
+                return (int) $value;
```

Comment:
What about the TYPE_ARRAY?

Reply:
If the value is an array I don't think we have to cast it 🤔. Is it possible to do this `return (array) $value` ? I thought that like a mixed type it should simply be returned as is.

But to be honest, I didn't need to use the type array for now so I'm not 100% sure, I added it because I thought it could be useful later. But maybe it would be better not to handle it at all for now until we encounter a use case where it is needed.

Reply:
We can cast things into array it works ^^
```
(array) 1; => [1]

Reply:
For record ^^https://github.com/PrestaShop/PrestaShop/blob/develop/tests/Integration/Behaviour/Features/Context/Util/PrimitiveUtils.php

Reply:
Ok I'll add the cast for array then